### PR TITLE
Use try-with-resources where possible

### PR DESCRIPTION
**Pull request by basil**
Before Java 9, we could only use fresh variables inside a try-with-resources block. As of Java 9 and as part of [JEP 213](https://openjdk.org/jeps/213), we can now use `final` or even effectively final variables inside a `try-with-resources` block. The benefit of try-with-resources is that they suppress exceptions thrown during close, propagating the original exception instead.

### Proposed changelog entries

N/A

### Proposed upgrade guidelines

N/A

### Submitter checklist

- [ ] (If applicable) Jira issue is well described
- [ ] Changelog entries and upgrade guidelines are appropriate for the audience affected by the change (users or developer, depending on the change) and are in the imperative mood. [Examples](https://github.com/jenkins-infra/jenkins.io/blob/master/content/_data/changelogs/weekly.yml)
  * Fill-in the `Proposed changelog entries` section only if there are breaking changes or other changes which may require extra steps from users during the upgrade
- [ ] Appropriate autotests or explanation to why this change has no tests
- [ ] New public classes, fields, and methods are annotated with `@Restricted` or have `@since TODO` Javadoc, as appropriate. 
- [ ] New deprecations are annotated with `@Deprecated(since = "TODO")` or `@Deprecated(forRemoval = true, since = "TODO")` if applicable.
- [ ] For dependency updates: links to external changelogs and, if possible, full diffs

### Desired reviewers

@mention

### Maintainer checklist

Before the changes are marked as `ready-for-merge`: 

- [ ] There are at least 2 approvals for the pull request and no outstanding requests for change
- [ ] Conversations in the pull request are over OR it is explicit that a reviewer does not block the change
- [ ] Changelog entries in the PR title and/or `Proposed changelog entries` are accurate, human-readable, and in the imperative mood
- [ ] Proper changelog labels are set so that the changelog can be generated automatically
- [ ] If the change needs additional upgrade steps from users, `upgrade-guide-needed` label is set and there is a `Proposed upgrade guidelines` section in the PR title. ([example](https://github.com/jenkinsci/jenkins/pull/4387))
- [ ] If it would make sense to backport the change to LTS, a Jira issue must exist, be a _Bug_ or _Improvement_, and be labeled as `lts-candidate` to be considered (see [query](https://issues.jenkins.io/issues/?filter=12146)).

<a href="https://gitpod.io/#https://github.com/jenkinsci/jenkins/pull/6765"></a>

### CODE REVIEW

**Review by NotMyFault — APPROVED** [review]

**Review by timja — APPROVED** [review]

File: cli/src/main/java/hudson/cli/CLI.java [hunk 1]
```diff
@@ -485,10 +485,8 @@ private static String computeVersion() {
         try {
             InputStream is = CLI.class.getResourceAsStream("/jenkins/cli/jenkins-cli-version.properties");
             if (is != null) {
-                try {
+                try (is) {
                     props.load(is);
-                } finally {
-                    is.close();
                 }
             }
```
Comment (jtnord) [inline cli/src/main/java/hudson/cli/CLI.java:491]: try-with-resources is `null` safe.

Why not

```
        try (InputStream is = CLI.class.getResourceAsStream("/jenkins/cli/jenkins-cli-version.properties")) {
            if (is != null) {
                    props.load(is);
                }
            }
```

File: core/src/main/java/hudson/FilePath.java [hunk 2]
```diff
@@ -533,10 +533,8 @@ private static class Archive extends MasterToSlaveFileCallable<Integer> {
         @Override
             public Integer invoke(File f, VirtualChannel channel) throws IOException {
                 Archiver a = factory.create(out);
-                try {
+                try (a) {
                     scanner.scan(f, ignoringSymlinks(a, verificationRoot, noFollowLinks));
-                } finally {
-                    a.close();
                 }
                 return a.countEntries();
```
Comment (jtnord) [inline core/src/main/java/hudson/FilePath.java:539]: would also seem more readable as:

```
                try (Archiver a = factory.create(out)) {
                    scanner.scan(f, ignoringSymlinks(a, verificationRoot, noFollowLinks));
                    return a.countEntries();
                }
```

Comment (basil) [reply core/src/main/java/hudson/FilePath.java:539]: Pre-existing issue

File: cli/src/main/java/hudson/cli/CLI.java (hunk 1 above)
Comment (basil) [reply cli/src/main/java/hudson/cli/CLI.java:491]: Pre-existing issue

**Comment by basil** [comment]
I agree that your suggestions are good improvements, but I am not going to fix pre-existing issues not caused by this PR.

**Comment by basil** [comment]
@jtnord I had some extra time on my day off and the changes seemed relatively simple, so I have accommodated your requests. In the future, kindly refrain from commenting on my PRs with requests for change regarding pre-existing issues.

**Review by jtnord — APPROVED** [review]

**Pull request merged**